bpf_test.go

```diff
@@ -253,11 +261,105 @@ func checkPinConfigCleanup(t *testing.T, expectedPaths []string) {
 	}
 }
 
+func checkUpdateDeleteElement(t *testing.T, b *elf.Module) {
+	mp := b.Map("dummy_hash")
+	if mp == nil {
+		t.Fatal("unable to find dummy_hash map")
+	}
+
+	key := 1000
+	value := 1000
+	if err := b.UpdateElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&value), BPF_ANY); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := b.UpdateElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&value), BPF_NOEXIST); err == nil {
+		t.Fatal(err)
+	}
+
+	if err := b.UpdateElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&value), BPF_EXIST); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := b.DeleteElement(mp, unsafe.Pointer(&key)); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := b.UpdateElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&value), BPF_EXIST); err == nil {
+		t.Fatal(err)
+	}
+}
+
+func checkLookupElement(t *testing.T, b *elf.Module) {
+	mp := b.Map("dummy_hash")
+	if mp == nil {
+		t.Fatal("unable to find dummy_hash map")
+	}
+
+	key := 2000
+	value := 2000
+	if err := b.UpdateElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&value), BPF_ANY); err != nil {
+		t.Fatal(err)
+	}
+
+	var lvalue int
+	if err := b.LookupElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&lvalue)); err != nil {
+		t.Fatal(err)
+	}
+	if value != lvalue {
+		t.Fatalf("wrong value returned, expected %d, got %d", value, lvalue)
+	}
+
+	key = 3000
+	if err := b.LookupElement(mp, unsafe.Pointer(&key), unsafe.Pointer(&lvalue)); err == nil {
+		t.Fatal(err)
+	}
+
+	expected := map[int]bool{2000: false}
```

Comment:
This map is a bit confusing with that name. I'd call this `found` and the `found` below just `f`.